scripts/generate-api-index.ts

```diff
@@ -0,0 +1,117 @@
+const { promises, writeFile, appendFile } = require('fs')
+const { exec } = require('child_process')
+
+type FileResult = {
+  name: string
+  path: string
+  apiName: string
+}
+
+type DirectoryResult = {
+  results: DirectoryResult[] | FileResult[]
+  group: string
+}
+
+const FILE = './src/api/index.ts'
+
+function camelize(str?: string) {
+  return str
+    ? str
+        .replace(/(?:^\w|[A-Z]|\b\w)/g, (word, index) => {
+          return index === 0 ? word.toLowerCase() : word.toUpperCase()
+        })
+        .replace(/\s+/g, '')
+    : null
+}
```

Comment:
I think this would be more readable without a tenary:

```suggestion
function camelize(str?: string) {
  if (!str) {
    return null
  }

  return str
    .replace(/(?:^\w|[A-Z]|\b\w)/g, (word, index) => index === 0 ? word.toLowerCase() : word.toUpperCase())
    .replace(/\s+/g, '')
}
```